src/g_compatibility.c

```diff
@@ -0,0 +1,333 @@
+//
+// Copyright(C) 2021 by Ryan Krafnick
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// DESCRIPTION:
+//  DSDA Compatibility
+//
+
+#include <stdio.h>
+
+#include "doomstat.h"
+#include "doomtype.h"
+#include "i_printf.h"
+#include "w_wad.h"
+
+#include "md5.h"
+
+typedef struct
+{
+    int comp;
+    int value;
+} option_t;
+
+typedef struct
+{
+    const char* cksum_string;
+    const option_t options[];
+} comp_record_t;
+
+#define END {-1, 0}
+
+static const comp_record_t eternal_doom_25 = {
+  "6da6fcba8089161bdec6a1d3f6c8d60f",
+  { {comp_stairs, 1}, {comp_vile, 1}, END }
+};
+
+// static const comp_record_t doomsday_of_uac_e1m8 = {
```

Comment:
Because it's not part of the MBF21 standard. "Doomsday of UAC" is nice, though.

Reply:
Currently, we would have to change the entire complevel for this to work properly, right?

Reply:
I think we would just have to switch this line from `demo_compatibility` to `comp[comp_666]`:

https://github.com/fabiangreffrath/woof/blob/ca4152176ce032946be5bd73363d489e77a72961/src/p_enemy.c#L2346

Reply:
Done in b15c2ad, but I'm not sure. Does it only work with non-Ultimate IWAD?

Reply:
Thank you so much! PrBoom+/DSDA-Doom check for `gameepisode < 4` instead of `gameversion`, I guess the latter check was from Chocolate Doom. Let's just follow suit.

Reply:
I think the default value for `comp[comp_666]` should depend on `gameversion`, though. 🤔